### DISCUSSION TITLE
Reference format for local Appimage on Linux ubuntu

### DISCUSSION THREAD

**Discussion by jo-mc** [GMOD/jbrowse-components#3013] [Q&A] [answered]
HI,
I can open and run the Appimage, however when I try and add my reference I get an error.  I suspect something other than a n.fa file and n.fa.fai file are required ?  The quickstart list => hg19...  hg38.. and mm10.. will open OK.

Version: /jbrowse-desktop-v1.7.9-linux.AppImage   ( I also tried 1.7.8)

My Steps
0. from my  fasta file generate a reference with samtools faidx : ref.fa & ref.fa.fai
1 Open sequence file > Indexed Fasta Adapter,  Fastalocation: local file "ref.fa" ; Fail location: local file "ref1.fa"  -> SUBMIT -> Launch View
2  See my assembly name and first feature name -> open -> Error

Error: [mobx-state-tree] Error while converting `{"displayedRegions":[{"refName":">L1HS","start":0,"end":null,"reversed":false,"assemblyName":"tes ref"}],"interRegionPaddingWidth":0,"minimumBlockWidth":3}` to `Base1DView`:

    at path "/displayedRegions/0/end" value `null` is not assignable to type: `number` (Value is not a number).
-------------------------------

Hopefully something simple,

thanks, Joe.

**Comment by cmdcolin** [comment, marked as answer]
> Open sequence file > Indexed Fasta Adapter, Fastalocation: local file "ref.fa" ; Fail location: local file "ref1.fa" 

I think it should be

FastaLocation: ref.fa
FaiLocation: ref.fa.fai

The error you are seeing is probably because you didn't put the fai into faiLocation

It might be possible to add a better error handling but hope that helps!

**Reply by jo-mc** [reply]
thanks, 
yes I did not put the fai in the failocation.  ( I assumed  the fai would be assumed, also I did not realize 'failocation" was after the fai file - obvious now)

**Reply by cmdcolin** [reply]
no worries! There is definitely an opportunity to improve the design a bit
